hearth/lib/hearth/event_stream/middleware/sign.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module Hearth
+  module EventStream
```

Comment:
Ditto on this living in EventStream. But also - why not make this functionality part of sign? Whether it signs event stream or not, it's still signing, and we can just make a branching condition there.

Reply:
Not sure if you saw this

Reply:
Thanks for the bump - this didn't show up in the conversation for some reason.

My original thought was that this is completely different from regular signing and I didn't want to add complexity to that middleware.  But I actually ended up pushing most of the complexity into the signer implementations, so I'll combine them back into one middleware.